test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectHandlerTests.cs

```diff
@@ -0,0 +1,766 @@
+                    app.Use(async (context, next) =>
+                    {
+                        await next();
+                    });
+                },
+                services =>
+                {
+                    services.AddDataProtection();
+                }
+            );
+        }
+    }
+
+    /// <summary>
+    /// Extension specifies <see cref="CustomOpenIdConnectAuthenticationMiddleware"/> as the middleware.
+    /// </summary>
+    public static class OpenIdConnectAuthenticationExtensions
+    {
+        /// <summary>
+        /// Adds the <see cref="OpenIdConnectAuthenticationMiddleware"/> into the ASP.NET runtime.
+        /// </summary>
+        /// <param name="app">The application builder</param>
+        /// <param name="customConfigureOption">Options which control the processing of the OpenIdConnect protocol and token validation.</param>
+        /// <param name="loggerFactory">custom loggerFactory</param>
+        /// <returns>The application builder</returns>
+        public static IApplicationBuilder UseCustomOpenIdConnectAuthentication(this IApplicationBuilder app, CustomConfigureOptions customConfigureOption, ILoggerFactory loggerFactory)
+        {
+            return app.UseMiddleware<CustomOpenIdConnectAuthenticationMiddleware>(customConfigureOption, loggerFactory);
+        }
+
+        /// <summary>
+        /// Adds the <see cref="OpenIdConnectAuthenticationMiddleware"/> into the ASP.NET runtime.
+        /// </summary>
+        /// <param name="app">The application builder</param>
+        /// <param name="options">Options which control the processing of the OpenIdConnect protocol and token validation.</param>
+        /// <param name="loggerFactory">custom loggerFactory</param>
+        /// <returns>The application builder</returns>
+        public static IApplicationBuilder UseCustomOpenIdConnectAuthentication(this IApplicationBuilder app, IOptions<OpenIdConnectAuthenticationOptions> options, ILoggerFactory loggerFactory)
+        {
+            return app.UseMiddleware<CustomOpenIdConnectAuthenticationMiddleware>(options, loggerFactory);
+        }
+    }
+
+    public class OpenIdConnectAuthenticationContext : IAuthenticateContext
+    {
+        public OpenIdConnectAuthenticationContext(string scheme = null)
+        {
+            AuthenticationScheme = scheme ?? OpenIdConnectAuthenticationDefaults.AuthenticationScheme;
+        }
+
+        public string AuthenticationScheme
+        {
+            get;
+            set;
+        }
+
+        public void Authenticated(ClaimsPrincipal principal, IDictionary<string, string> properties, IDictionary<string, object> description)
+        {
+        }
+
+        public void NotAuthenticated()
+        {
+        }
+    }
+
+    /// <summary>
+    /// Provides a Facade over IOptions
+    /// </summary>
+    public class Options : IOptions<OpenIdConnectAuthenticationOptions>
```

Comment:
Why do you need this?  You should be able to use the real options, or you could use Moq